apiserver/facades/controller/charmrevisionupdater/testing/suite.go

```diff
@@ -84,14 +84,17 @@ func (s *CharmSuite) AddMachine(c *gc.C, machineId string, job state.MachineJob)
 		Series: "quantal",
 		Jobs:   []state.MachineJob{job},
 	})
+
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(m.Id(), gc.Equals, machineId)
 	cons, err := m.Constraints()
 	c.Assert(err, jc.ErrorIsNil)
 	controllerCfg, err := s.jcSuite.State.ControllerConfig()
 	c.Assert(err, jc.ErrorIsNil)
 	inst, hc := jujutesting.AssertStartInstanceWithConstraints(c, s.jcSuite.Environ, s.jcSuite.ProviderCallContext, controllerCfg.ControllerUUID(), m.Id(), cons)
-	err = m.SetProvisioned(inst.Id(), "fake_nonce", hc)
+	displayName, err := m.DisplayName()
```

Comment:
This isn't right - you are setting the displayname of a machine to the displayname of a machine.
The display name comes from the act of starting an instance - so the AssertStartInstanceWithConstraints() helper above needs to return  an inst implementation that has the display name added

Reply:
Fixed